src/zarr/core/sync.py

```diff
@@ -36,6 +38,45 @@ class SyncError(Exception):
     pass
 
 
+# TODO: not sure if there is a better place to put this function, so I've kept it here for now.
+def _make_shutdown_asyncgens_noop_for_pyodide() -> None:
+    """
+    Patch Pyodide's WebLoop to fix interoperability with pytest-asyncio.
+
+    WebLoop.shutdown_asyncgens() raises NotImplementedError, which causes
+    pytest-asyncio to issue warnings during test cleanup and potentially
+    cause resource leaks that make tests hang. This is a bit of a
+    hack, but it allows us to run tests that use pytest-asyncio.
+
+    This is necessary because pytest-asyncio tries to clean up async generators
+    when tearing down test event loops, but Pyodide's WebLoop doesn't support
+    this as it integrates with the browser's event loop rather than managing
+    its own lifecycle.
+    """
+    try:
+        if not IS_WASM and "pyodide" not in sys.modules:
+            return
```

Comment:
The missing coverage annotations for this test are because Codecov does not know about this WASM CI job where these code paths are hit. I can mark them with a `# pragma: no cover` comment closer to the time of merging this PR.